ys4512558/Week_16/BOJ_21924_도시건설/BOJ_21924_도시건설.java

```diff
@@ -0,0 +1,84 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.PriorityQueue;
+import java.util.StringTokenizer;
+
+public class BOJ21924 {
+    static int[] parents;
+    static int[] rank;
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+
+        int N = Integer.parseInt(st.nextToken());
+        int M = Integer.parseInt(st.nextToken());
+        parents = new int[N + 1];
+        rank = new int[N + 1];
+        for (int i = 1; i <= N; i++) {
+            parents[i] = i;
+            rank[i] = 1;
+        }
```

Comment:
rank 배열을 따로 만들지 않고, parents 배열에다가 양수는 **현재 노드의 부모**로, 음수는 **현재 집합의 크기**로 지정하는 방법도 있습니다! 초기값은 부모가 없고, 집합의 크기가 1이기 때문에 아래처럼 설정할 수 있겠네요.
```suggestion
        parents = new int[N + 1];
        for (int i = 1; i <= N; i++) {
            parents[i] = -1;
        }
```